R/write.R

```diff
@@ -508,18 +508,35 @@ check_labelled <- function(supertbl, add_labelled_column_headers, call = caller_
 supertbl_recode <- function(supertbl, supertbl_meta) {
   # Recode yesno from TRUE/FALSE to "yes"/"no"
 
-  yesno_fields <- supertbl_meta %>%
+  yesno_fields <- supertbl_meta %>% # nolint: object_usage_linter
     filter(.data$field_type == "yesno") %>%
     pull(.data$field_name)
 
-  checkbox_fields <- supertbl_meta %>%
+  checkbox_fields <- supertbl_meta %>% # nolint: object_usage_linter
     filter(.data$field_type == "checkbox") %>%
     pull(.data$field_name)
 
-  map(
+  # Recode logical vars, define and re-apply labels (similar to labelled.R)
+  # as these are lost during attribute changes
+  map2(
```

Comment:
Maybe I'm missing something here but I don't think we want to rebuild the labels from scratch. Just save the existing labels, do the recoding, then restore the existing labels we saved.

Reply:
I think this most recent commit should address this

Reply:
@ezraporter lmk what you think in most recent commit

Reply:
Looks good to me!